Client/Frontend/Login Management/LoginListViewController.swift

```diff
@@ -177,18 +178,20 @@ class LoginListViewController: SensitiveViewController {
     }
 
     private func reloadTableWithResult(result: Maybe<Cursor<Login>>) -> Success {
-        loadingStateView.hidden = true
-        loginDataSource.allLogins = result.successValue?.asArray() ?? []
-        tableView.reloadData()
-        activeLoginQuery = nil
+        let logins = result.successValue?.asArray() ?? []
+        return loginDataSource.setData(logins).bindQueue(dispatch_get_main_queue()) { _ in
+            self.loadingStateView.hidden = true
+            self.tableView.reloadData()
+            self.activeLoginQuery = nil
+
+            if self.loginDataSource.count > 0 {
+                self.navigationItem.rightBarButtonItem?.enabled = true
+            } else {
+                self.navigationItem.rightBarButtonItem?.enabled = false
+            }
```

Comment:
While here, could simplify to `self.navigationItem.rightBarButtonItem?.enabled = (self.loginDataSource.count > 0)` if you want.